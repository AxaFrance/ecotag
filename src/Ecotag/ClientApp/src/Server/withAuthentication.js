import React from 'react';
import {useOidcAccessToken, useOidcIdToken, useOidcUser} from '@axa-fr/react-oidc';
import {EnvironmentConsumer} from './EnvironmentProvider';

const NON_CONNECTE = 'Non ConnectÃ©';

const getAuthName = (dataArray, propertyName = "name", defaultResult = NON_CONNECTE) => {

    for (let i = 0; i < dataArray.length; i++) {
        const item = dataArray[i];
        if (item && item[propertyName]) {
            return item[propertyName];
        }
    }

    return defaultResult;
}

export const DataScientist = "ECOTAG_DATA_SCIENTIST";
export const Annotateur = "ECOTAG_ANNOTATEUR";
export const Administateur = "ECOTAG_ADMINISTRATEUR";

const addRole = (roles, role) => {
    if (roles.includes(role)) {
        return;
    }
    if (role === DataScientist || role === Annotateur || role === Administateur) {
        roles.push(role);
    }
}

export const extractRoles = (accessTokenPayload, oidcMode) => {
    const roles = [];
    if (oidcMode === "AXA_FRANCE") {
        if (accessTokenPayload && accessTokenPayload.member_of && accessTokenPayload.member_of.length > 0) {
              accessTokenPayload.forEach((item) => addRole(roles, item));
            if (roles.includes(DataScientist)) {
                addRole(roles, Annotateur);
            } else if (roles.includes(Administateur)) {
                addRole(roles, DataScientist)
                addRole(roles, Annotateur)
            }
        }
    } else {
        roles.push(Annotateur);
        roles.push(DataScientist);
        roles.push(Administateur);
    }
    return roles;
}

const extractDataFromOAuthToken = (idTokenPayload, accessTokenPayload, oidcUser, environment) => ({
    name: getAuthName([idTokenPayload, accessTokenPayload]),
    email: getAuthName([oidcUser, accessTokenPayload], 'email', ''),
    roles: extractRoles(accessTokenPayload, environment.oidc.mode)
});

const withAuthentication = () => Component => props => {
    const {idTokenPayload} = useOidcIdToken();
    const {accessTokenPayload} = useOidcAccessToken();
    const {oidcUser, oidcUserLoadingState} = useOidcUser();
    return <EnvironmentConsumer>{store => <Component {...props}
                                                     user={extractDataFromOAuthToken(idTokenPayload, accessTokenPayload, oidcUser, store.environment)}
                                                     UserLoadingState={oidcUserLoadingState}/>}</EnvironmentConsumer>;
};

export default withAuthentication;
