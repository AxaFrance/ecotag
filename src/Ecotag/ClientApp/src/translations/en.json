{
  "header": {
    "title": "An eco-system that aims to be eco-friendly",
    "errors": {
      "not_connected": "Not connected"
    }
  },
  "paging": {
    "display": "Display",
    "elements": "elements",
    "previous": "« Previous",
    "next": "Next »"
  },
  "stepper": {
    "confirm": "Confirm"
  },
  "resilience": {
    "error": {
      "first_part": "An error occurred.",
      "second_part": "We are aware of the problem. An investigation is ongoing..."
    },
    "forbidden": "You are not allowed to access this resource."
  },
  "not_found": {
    "title": "404 Page Not Found",
    "not": "not",
    "found": "found",
    "message": "The page you are looking for is not here !"
  },
  "home": {
    "title": "Home",
    "projects": "Projects",
    "datasets": "Datasets",
    "groups": "Teams",
    "errors": {
      "no_roles": "No role is associated to your profile."
    }
  },
  "group": {
    "title": "Teams",
    "subtitle": "Annotation teams governance",
    "creation_title": "Creation",
    "new": {
      "name_input": {
        "label": "Name : ",
        "help_message": "Ex : bank account details annotators"
      },
      "validation": "Validate"
    },
    "count_teams": {
      "first_part": "There is/are (",
      "second_part": ") team(s)"
    },
    "empty_list": "No element",
    "list": {
      "header": {
        "name": "Name",
        "users": "Users",
        "actions": "Actions"
      },
      "edit": "Edit",
      "edit_popup": {
        "header": "Add/Remove users from this team",
        "add_user_placeholder": "Add user",
        "footer": {
          "cancel": "Cancel",
          "validation": "Validate"
        }
      }
    }
  },
  "dataset": {
    "title": "Datasets",
    "subtitle": "A dataset is the representation of a collection of data",
    "new_dataset": "New dataset",
    "access_token": "Access token",
    "active_datasets_counter": "Your active datasets : ",
    "filter_by": "Filter by",
    "name": "Name",
    "empty_list": "No elements",
    "list": {
      "actions_titles": {
        "locked": "locked",
        "locked_and_work_in_progress": "locked and work in progress",
        "pending": "pending",
        "none": "unlocked"
      },
      "header": {
        "lock": "Lock",
        "name": "Name",
        "team": "Team",
        "classification": "Classification",
        "nb_files": "Number of files",
        "type": "Type",
        "creation_date": "Creation date",
        "action": "Action"
      },
      "edit": "Edit"
    },
    "new": {
      "title": "New dataset",
      "stepper": {
        "title": "New dataset"
      },
      "form_title": "New dataset",
      "name_input": {
        "label": "Name",
        "help_message": "Ex : bank-account-details-dataset"
      },
      "team_input": {
        "label": "Team"
      },
      "type_input": {
        "label": "Type",
        "help_message": "Ex : Image, Text",
        "options_labels": {
          "image": "Images (.jpg, .png, jpeg)",
          "document": "Documents (.txt, .eml, .tiff, .pdf, .doc, etc.)",
          "text": "Text (.txt)",
          "eml": "Emails (.eml)"
        },
        "help": {
          "title": "Information on the accepted documents types",
          "images": {
            "name": "Images",
            "types": " : .jpg, .png, jpeg"
          },
          "documents": {
            "name": "Documents",
            "types": " : .txt, .eml, .msg, .jpg, .png, jpeg, .tiff, .tif, .pdf, .doc, .docx, .xls, .xlsx, .ppt, .pptx, .rtf, .odt, .ods, .odp, .zip"
          },
          "text": {
            "name": "Text",
            "types": " : .txt"
          },
          "eml": {
            "name": "Eml",
            "types": " : .eml"
          }
        }
      },
      "classification_input": {
        "label": "Classification",
        "help_message": "Ex : Critical",
        "options": {
          "public": {
            "label": "Public"
          },
          "internal": {
            "label": "Internal"
          },
          "confidential": {
            "label": "Confidential"
          },
          "critical": {
            "label": "Critical"
          }
        },
        "help": {
          "title": "Information on criticality",
          "public": {
            "name": "Public",
            "detail": " : a piece of information made deliberately for the public, or specifically made for public diffusion"
          },
          "internal": {
            "name": "Internal",
            "detail": " : a piece of information that is accessible to all or a large portion of employees and is not intended for external parties."
          },
          "confidential": {
            "name": "Confidential",
            "detail": " : a piece of information restricted to a specific audience"
          },
          "critical": {
            "name": "Critical",
            "detail": " : Please refer to your manager."
          }
        }
      },
      "import_input": {
        "label": "Import from a blob : "
      },
      "dataset_to_import": {
        "label": "Dataset to import : "
      },
      "validate": "Validation"
    },
    "confirm": {
      "title": "Confirmation",
      "stepper": "New dataset",
      "confirmation_title": "New dataset added !",
      "confirmation_description": "If you have imported a dataset, the process of adding files can take a while.",
      "return_button_label": "Go back to datasets list"
    },
    "edit": {
      "confirm_modal": {
        "title": "Do you want to lock the dataset permanently ?",
        "warning_first_part": "This action is permanent.",
        "warning_second_part": "Every modification (addition or deletion of files) will be impossible then."
      },
      "subtitle": "Dataset edition",
      "lock": "Lock",
      "locked_state": {
        "locked": "Dataset locked",
        "locked_and_work_in_progress": "Dataset is being worked on",
        "default": "Dataset is being imported"
      },
      "file_upload": {
        "title": "Upload files ",
        "file_input": {
          "label": "Browse",
          "placeholder": "Drag/drop your files here"
        },
        "send_button": {
          "label": "Send"
        },
        "errors": {
          "already_existing": {
            "first_part": "The file ",
            "second_part": " already exists"
          },
          "could_not_upload": {
            "first_part": "The file ",
            "second_part": " could not be uploaded"
          }
        }
      },
      "files_list": {
        "global_info_tab": {
          "title": "Global information",
          "name": "Name",
          "creation_date": "Creation date",
          "type": "Type",
          "nb_files": "Number of files",
          "files_total_size": "Total files size",
          "files_mean_size": "Average file size"
        },
        "files_list_tab": {
          "title": "Files list",
          "name": "Name",
          "type": "Type",
          "size": "Size",
          "action": "Action",
          "delete_action": "Delete"
        }
      }
    }
  }
}