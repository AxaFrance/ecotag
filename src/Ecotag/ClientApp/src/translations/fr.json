{
  "header": {
    "title": "Un écosystème qui vise à être écologique",
    "errors": {
      "not_connected": "Non connecté"
    }
  },
  "home": {
    "title": "Accueil",
    "projects": "Projets",
    "datasets": "Datasets",
    "groups": "Groupes",
    "errors": {
      "no_roles": "Aucun rôle n'est attribué à votre profil."
    }
  }
}