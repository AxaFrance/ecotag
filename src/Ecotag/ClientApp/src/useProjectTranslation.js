import './i18n';
import {useTranslation} from "react-i18next";
import i18next from 'i18next';

const useProjectTranslation = (namespace = 'translation') => {
    const {t: translate} = useTranslation(namespace);
    return {translate};
};

export const changeProjectTranslationLanguage = (language) => {
    i18next.changeLanguage(language);
};

export default useProjectTranslation;
