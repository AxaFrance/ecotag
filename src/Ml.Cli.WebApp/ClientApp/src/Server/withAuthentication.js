import React from 'react';
import {useOidcAccessToken, useOidcIdToken, useOidcUser} from '@axa-fr/react-oidc';
import { EnvironmentConsumer } from './EnvironmentProvider';

const NON_CONNECTE = 'Non ConnectÃ©';

const getAuthName = (dataArray, propertyName= "name", defaultResult=NON_CONNECTE) => {

    for(let i=0; i<dataArray.length;i++) {
        const item= dataArray[i];
        if (item && item[propertyName]) {
            return item[propertyName];
        }
    }

    return defaultResult;
}

export const DataScientist = "ECOTAG_DATA_SCIENTIST";
export const Annotateur = "ECOTAG_ANNOTATEUR";
export const Administateur = "ECOTAG_ADMINISTRATEUR";
    
export const extractRoles = (oidcUser, oidcMode) => {
    const roles = [];
    if(oidcMode === "AXA_FRANCE"){
          if(oidcUser && oidcUser.member_of && oidcUser.member_of.length > 0) {
              oidcUser.member_of.forEach(member => {
                  member.split(",").forEach(subMember => {
                      if(subMember.startsWith('CN=')){
                          const role = subMember.replace('CN=', '');
                          if(role.includes("ECOTAG_")) {
                              roles.push(role);
                          }
                      }
                  });
              });
              if(roles.includes(DataScientist)){
                  roles.push(Annotateur);
              }
              else if(roles.includes(Administateur)){
                  roles.push(DataScientist);
                  roles.push(Annotateur);
              }
          }
    }else{
        roles.push(Annotateur);
        roles.push(DataScientist);
        roles.push(Administateur);
    }
  return roles;
}

const extractDataFromOAuthToken = (idTokenPayload, accessTokenPayload, oidcUser, environment) => ({
  name: getAuthName([idTokenPayload, accessTokenPayload]),
  email: getAuthName([oidcUser, accessTokenPayload], 'email', ''),
  roles: extractRoles(oidcUser, environment.oidc.mode)
});

const withAuthentication = () => Component => props => {
  const { idTokenPayload } = useOidcIdToken();
  const { accessTokenPayload } = useOidcAccessToken();
  const{ oidcUser, oidcUserLoadingState } = useOidcUser();
  return <EnvironmentConsumer>{store =><Component {...props} user={extractDataFromOAuthToken(idTokenPayload, accessTokenPayload, oidcUser, store.environment)} UserLoadingState={oidcUserLoadingState} />}</EnvironmentConsumer>;
};

export default withAuthentication;
