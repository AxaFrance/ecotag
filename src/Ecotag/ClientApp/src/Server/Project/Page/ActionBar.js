import React from 'react';
import {useHistory} from 'react-router-dom';
import Button from '@axa-fr/react-toolkit-button';
import './Page.scss';
import ExportButton from "./ExportButton";
import useProjectTranslation from "../../../translations/useProjectTranslation";

export const ActionBar = ({projectId, projectName, isAnnotationClosed, onExport, user}) => {
    const history = useHistory();
    const {translate} = useProjectTranslation();

    const startTaggingButton = () => {
        const path = `/projects/${projectId}/annotations/start`;
        history.push(path);
    };

    if (!projectId) {
        return null;
    }

    return (
        <div className="ft-actionBar">
            {(isAnnotationClosed) ? null : <Button onClick={startTaggingButton} id="startTagging" name={translate('project.project_page.action_bar.start_button_label')}>
                <span className="af-btn-text">{translate('project.project_page.action_bar.start_button_label')}</span>
            </Button>}
            <ExportButton user={user} projectId={projectId} projectName={projectName} onExport={onExport}/>
        </div>
    );
};

export default ActionBar;
