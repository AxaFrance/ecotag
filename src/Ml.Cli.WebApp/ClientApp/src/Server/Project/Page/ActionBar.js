import React from 'react';
import { useHistory } from 'react-router-dom';

import Button from '@axa-fr/react-toolkit-button';
import './Page.scss';

const showEditButton = isAdmin => {
  return isAdmin ? (
    <div>
      <Button className="btn af-btn af-btn--success af-btn--hasiconRight" tabIndex="-1">
        <span className="af-btn__text">Edit Project</span>
        <i className="glyphicon glyphicon-pencil"/>
      </Button>
    </div>
  ) : null;
};

export const ActionBar = ({ currentUser, project, dataset }) => {
  const history = useHistory();

  const startTaggingButton = () => {
    const path = `/projects/${project.id}/annotations/${dataset.id}`;
    history.push(path);
  };

  return (
    <div className="ft-actionBar">
      {showEditButton(currentUser.role === 'ADMIN')}
      <div>
        <Button onClick={() => startTaggingButton()} id="startTagging" name="Start Tagging">
          <span className="af-btn-text">Start Tagging</span>
        </Button>
      </div>
    </div>
  );
};

export default ActionBar;
