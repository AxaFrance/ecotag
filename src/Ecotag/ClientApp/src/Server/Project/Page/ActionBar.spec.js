import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import {render, waitFor} from '@testing-library/react';
import {ActionBar} from './ActionBar';
import {BrowserRouter as Router} from "react-router-dom";
import {DataScientist} from "../../withAuthentication";
import {changeProjectTranslationLanguage} from "../../../translations/useProjectTranslation";

describe('ActionBar', () => {
    const renderComponentAndCheckSnapshot = async(expectedText) => {
        const regex = new RegExp(expectedText, 'i');
        const {asFragment, getByText} = render(<Router><ActionBar user={{roles: [DataScientist]}} projectId={"0001"}
                                                                  projectName={"projectName"}
                                                                  isAnnotationClosed={false}/></Router>);
        const messageEl = await waitFor(() => getByText(regex));
        expect(messageEl).toHaveTextContent(
            expectedText
        );
        expect(asFragment()).toMatchSnapshot();
    };

    it('should render correctly in english', async () => {
        changeProjectTranslationLanguage('en');
        await renderComponentAndCheckSnapshot('Start tagging');
    });
    it('should render correctly in french', async() => {
        changeProjectTranslationLanguage('fr');
        await renderComponentAndCheckSnapshot('Commencer Ã  annoter');
    })
});
