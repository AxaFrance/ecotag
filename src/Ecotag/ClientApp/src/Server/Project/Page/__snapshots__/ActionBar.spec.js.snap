// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionBar should render correctly in english 1`] = `
<DocumentFragment>
  <div
    class="ft-actionBar"
  >
    <button
      class="btn af-btn"
      id="startTagging"
      name="Start tagging"
    >
      <span
        class="af-btn-text"
      >
        Start tagging
      </span>
    </button>
    <a
      class="ft-actionBar__link"
      href=""
    >
      Export
    </a>
  </div>
</DocumentFragment>
`;

exports[`ActionBar should render correctly in french 1`] = `
<DocumentFragment>
  <div
    class="ft-actionBar"
  >
    <button
      class="btn af-btn"
      id="startTagging"
      name="Commencer à annoter"
    >
      <span
        class="af-btn-text"
      >
        Commencer à annoter
      </span>
    </button>
    <a
      class="ft-actionBar__link"
      href=""
    >
      Exporter
    </a>
  </div>
</DocumentFragment>
`;
